### PR TITLE
Use session for sts boto3 client for logging hook

### PR DESCRIPTION
**Pull request by tiffzhao5**
# Pull Request Summary

_What is this PR changing? Why is this change being made? Any caveats you'd like to highlight? Link any relevant documents, links, or screenshots here if applicable._

Allows boto3 to be thread safe, prevents errors like [these](https://app.datadoghq.com/logs?query=service%3Async-cheating-detection-v6-endpoint%20-status%3A%28info%20OR%20warn%29%20&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&event=AgAAAZDrvYuwpckYqQAAAAAAAAAYAAAAAEFaRHJ2WnlyQUFBV0xNRzB4OG1KTFFBRgAAACQAAAAAMDE5MGViYmYtZDgzZC00MTdiLWIyOGMtZjkyYTk4MjFmY2Qy&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=time%2Cdesc&view=spans&viz=stream&from_ts=1721841891918&to_ts=1721928291918&live=true):

image

## Test Plan and Usage Guide

_How did you validate that your PR works correctly? How do you run or demo the code? Provide enough detail so a reviewer can reasonably reproduce the testing procedure. Paste example command line invocations if applicable._

Shelled into pod of launch endpoint to test

### CODE REVIEW

File: model-engine/model_engine_server/inference/infra/gateways/firehose_streaming_storage_gateway.py [hunk 1]
```diff
@@ -29,7 +29,8 @@ def __init__(self):
     """
 
     def _get_firehose_client(self):
-        sts_client = boto3.client("sts", region_name=infra_config().default_region)
+        sts_session = boto3.Session(region_name=infra_config().default_region)
```
Comment (yunfeng-scale) [inline model-engine/model_engine_server/inference/infra/gateways/firehose_streaming_storage_gateway.py:32]: possible to create this client at the process level?

**Review by yunfeng-scale — APPROVED** [review]

**Review by dmchoiboi — APPROVED** [review]

File: model-engine/model_engine_server/inference/infra/gateways/firehose_streaming_storage_gateway.py (hunk 1 above)
Comment (tiffzhao5) [reply model-engine/model_engine_server/inference/infra/gateways/firehose_streaming_storage_gateway.py:32]: sorry what do you mean by "process level"?

Comment (yunfeng-scale) [reply model-engine/model_engine_server/inference/infra/gateways/firehose_streaming_storage_gateway.py:32]: like using a global variable

Comment (tiffzhao5) [reply model-engine/model_engine_server/inference/infra/gateways/firehose_streaming_storage_gateway.py:32]: I don't reuse it anywhere else actually so doesn't need to be global

also found this lol: https://stackoverflow.com/questions/19158339/why-are-global-variables-evil

**Pull request merged**